index.js

```diff
@@ -51,6 +51,23 @@ instance.prototype.incomingData = function(data) {
 		self.log('error', "incorrect username/password (expected no password)");
 		self.status(self.STATUS_ERROR, 'Incorrect user/pass');
 	}
+	// Heatbeat to keep connection alive (Working)
+	if (self.login === true && self.socket.connected) {
+		clearInterval(self.heartbeat_interval);
+		var beat_period = 180; // Seconds
+		self.heartbeat_interval = setInterval(heartbeat, beat_period * 1000);
+		function heartbeat() {
+			self.login = false;
+			self.status(self.STATUS_WARNING,'Checking Connection');
+			self.socket.write("1I"+ "\n"); // should respond with Switcher description (short) eg: Inf01*SMX
+			debug("Checking Connection");
+		}
+	}
+	else if (self.login === false && data.match("SMX")) {
```

Comment:
Code duplication. change the "V" search on line 45 to have both search parameters. Like `else if (self.login === false && data.match(/V|SMX/)) {`

Reply:
I did this to have a different debug message to identify heartbeat. Can I still achieve this?